### DISCUSSION TITLE
Connect to wezterm workspace from non-wezterm client?

### DISCUSSION THREAD

**Discussion by jaidetree** [wezterm/wezterm#4860] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

My current workflow is to run my editor and terminal windows (currently kitty) on my work computer then ssh into it from a tablet to work remotely through a tmux session.

Is it possible to connect to a wezterm workspace over ssh and use it like tmux? After trying to ssh in and run `wezterm connect local` it just tails the logs and opens a gui window I'm not able to see or interact with.

Is there a command I'm missing to get it to work as a cli or tui over ssh or is wezterm's multiplexing only aiming to support wezterm clients?

**Comment by wez** [comment, marked as answer]
At this time, the only ui for wezterm's mux protocol is in the wezterm-gui client process.

Now, I could potentially build a very basic tui client for these sorts of scenarios, but given that the multiplexer userbase tends to be predominantly power users, it wouldn't take long before the set of features there would need to be equivalent to tmux or zellij for it to be worthwhile for anyone to use it, and therefore to justify the effort in building it. There's a lot of work there, for something that I wouldn't be using myself -> I have low motivation to start or maintain such a thing.

Perhaps a better outcome would be to come up with a standard protocol that various multiplexer front ends could talk to, and then have tmux, zellij and/or others implement that? I do have thoughts about such a thing.

**Reply by jaidetree** [reply]
Thanks for the answer, that's totally fair! I don't blame you for not wanting to create another tmux or zellij, it would be a big undertaking without a lot of value that only diverts your attention from the main problem space of this project.

Creating a standard protocol does sound like a more worthy investment! Especially if existing clients like tmux could adopt it and if not, it at least opens the door for new clients from people that may want to build it.

Seems like I can't drop tmux from my toolchain yet but I'm still going to give wezterm a shot! Was able to bootstrap fennel quickly and the idea of customizing a terminal with all the lispy goodness I desire is very compelling to keep with it -- and as far as I can tell tmux does run just fine in it for now 🙂

**Comment by depeo** [comment]
This would be a great feature, sometimes I'd like to check up on progress of a certain task, on Android, when I'm away from the computer.
